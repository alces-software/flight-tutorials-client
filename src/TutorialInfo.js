// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

import React from 'react';
import type { TutorialType } from './types';

const TutorialInfo = ({
  tutorial,
} : {
  tutorial: TutorialType
}) => (
  <div>
    <h2>{tutorial.title}</h2>
    <div dangerouslySetInnerHTML={{ __html: tutorial.description }} />
  </div>
);

export default TutorialInfo;
