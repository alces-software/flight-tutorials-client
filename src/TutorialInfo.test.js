/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Compute Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TutorialInfo from './TutorialInfo'

const tutorial = {
  title: 'My tutorial',
  description: 'My tutorial description',
};

it('renders without crashing', () => {
  const node = document.createElement('div');
  render(<TutorialInfo tutorial={tutorial} />, node);
});

it('renders correctly', () => {
  const tree = renderer.create(
    <TutorialInfo tutorial={tutorial} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the title', () => {
  const wrapper = shallow(
    <TutorialInfo tutorial={tutorial} />
  );

  expect(wrapper).toIncludeText(tutorial.title);
});


// Add this test back in once we support markdown rendering of the
// description.
xit('renders the description', () => {
  const wrapper = shallow(
    <TutorialInfo tutorial={tutorial} />
  );

  expect(wrapper).toIncludeText(tutorial.description);
});
