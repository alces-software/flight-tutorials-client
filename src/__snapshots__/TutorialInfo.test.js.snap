// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Tutorial 1
  </h2>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "Tutorial 1 description",
      }
    }
  />
</div>
`;
